addons/a11y/src/index.ts

```diff
@@ -39,20 +39,29 @@ const run = (element: ElementContext, config: Spec, options: RunOptions) => {
             restoreScroll: true,
           } as RunOptions) // cast to RunOptions is necessary because axe types are not up to date
       )
-      .then(report);
+      .then(report)
+      .catch(error => addons.getChannel().emit(EVENTS.ERROR, String(error)));
   });
 };
 
 if (module && module.hot && module.hot.decline) {
   module.hot.decline();
 }
 
+let storedDefaultSetup: Setup | null = null;
+
 export const withA11y = makeDecorator({
   name: 'withA11Y',
   parameterName: PARAM_KEY,
   wrapper: (getStory, context, { parameters }) => {
     if (parameters) {
-      setup = parameters as Setup;
+      if (storedDefaultSetup === null) {
+        storedDefaultSetup = { ...setup };
+      }
+      Object.assign(setup, parameters as Setup);
+    } else if (storedDefaultSetup !== null) {
+      Object.assign(setup, storedDefaultSetup);
```

Comment:
Sorry for the question, but I don't really understand the goal. Isn't the issue about parameters isolation ?

AFAIU, here, when you add parameters, they are stored and become the default parameters.
But next time, when you don't pass any parameters, you use the default one, and reset the this default. Doesn't your previous parameters impact your current stories ?

And everytime, setup is changed, so on `EVENTS.REQUEST`, the arrow function uses the same setup ref, which content changes.

Am I missing something ?